src/shared/components/entry/EntryDownload.tsx

```diff
@@ -125,31 +218,145 @@ export type EntryDownloadProps = {
     downloadMethod?: DownloadMethod
   ) => void;
   columns?: Column[];
+  dataset?: Dataset;
+  featureTypes?: string[];
 };
 
 const EntryDownload = ({
   nResults,
   isoformsAvailable,
   onClose,
   columns,
+  dataset,
+  featureTypes,
 }: EntryDownloadProps) => {
   const match = useRouteMatch<{ namespace: Namespace; accession: string }>(
     allEntryPages
   );
   const { namespace, accession } = match?.params || {};
 
   const [downloadColumns, setDownloadColumns] = useState(columns);
-
-  const fileFormatEntryDownload = namespace && formatMap.get(namespace);
+  const [fileFormats, setFileFormats] = useState(
+    namespace && formatMap.get(namespace)
+  );
+  const [selectedFormat, setSelectedFormat] = useState(fileFormats?.[0]);
+  const [selectedDataset, setSelectedDataset] = useState(
+    dataset || Dataset.uniprotData
+  );
+  const [extraContent, setExtraContent] = useState('');
 
   const { data } = useDataApi<ReceivedFieldData>(
     namespace &&
       namespace !== Namespace.uniparc &&
-      (fileFormatEntryDownload?.includes(FileFormat.tsv) ||
-        fileFormatEntryDownload?.includes(FileFormat.excel))
+      (fileFormats?.includes(FileFormat.tsv) ||
+        fileFormats?.includes(FileFormat.excel))
       ? apiUrls.resultsFields(namespace)
       : null
   );
+  const { data: resultFieldsData } = useDataApi<ReceivedFieldData>(
+    namespace === Namespace.uniprotkb ? apiUrls.resultsFields(namespace) : null
+  );
+
+  const uniprotFeaturesMap = useMemo(() => {
+    const featuresMap = new Map();
+    if (resultFieldsData) {
+      const fields = resultFieldsData.flatMap((item) => item.fields);
+      fields.forEach((element) => {
+        if (element.name.startsWith('ft_')) {
+          // There is a mismatch in the feature type returned by the backend between entry and result-field endpoints.
+          // Two exceptions are Domain and Signal (in result fields they are named as 'Domain [FT]' and 'Signal peptide')
+          if (element.name === 'ft_domain') {
+            featuresMap.set('Domain', element.name);
+          } else if (element.name === 'ft_signal') {
+            featuresMap.set('Signal', element.name);
+          } else {
+            featuresMap.set(element.label, element.name);
+          }
+        }
+      });
+    }
+    return featuresMap;
+  }, [resultFieldsData]);
+
+  const availableDatasets = [Dataset.features];
+
+  const { data: geneCentricData } = useDataApi<SearchResults<GeneCentricData>>(
+    namespace === Namespace.uniprotkb && accession
+      ? apiUrls.genecentric(accession)
+      : ''
+  );
+
+  const proteinsApiVariation = useDataApi(
+    namespace === Namespace.uniprotkb
+      ? accession && joinUrl(proteinsApi.variation(accession))
```

Comment:
```suggestion
    namespace === Namespace.uniprotkb && accession ? joinUrl(proteinsApi.variation(accession))
```
I think this is a little clearer - the same comment applies to the user `useDataApi` invocations in this file.